tutorial/01_Introduction/intro.ipynb

```diff
@@ -8,7 +8,7 @@
    "source": [
```

Comment:
<p>I would not have the "shortcut" methods. Also would remove the "create" method.</p><p> </p><p>When an attribute is accessed the first time, in the setter, trigger the creation step.</p><p>It also would be useful to mention in the docstring that "flags", "boresight_radec" and "times" are special keys and when they are created, launch a warning if they do not respect the expected shape and dtype.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/hpc4cmb/toast/pull/369/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Reply:
<p>Thanks for your feedback here.  I think I need to include a broader discussion of naming conventions.  Basically every operator can use arbitrary names for every data product.  For example, the name "boresight_radec" is a trait value of the SimSatellite class.</p><p> </p><p>Each operator has methods to communicate the names of data objects that the operator "provides" and the names that it "requires".  So when running a <code>Pipeline</code>,  toast can check if all the operators are consistent and that all the inputs for every operator are provided by some earlier operator in the sequence.</p><p> </p><p>The setitem API does not support additional kwargs, so there is no way to create the data buffer with a custom shape and dtype on first access.</p><p> </p>

Reply:
<p>Same comment here about setitem / create method.  If it is more intuitive, I could accept a tuple or dictionary to setitem to pass in all the parameters.</p>

Reply:
> The setitem API does not support additional kwargs, so there is no way to create the data buffer with a custom shape and dtype on first access.

    def __setitem__(self, key, value):
        dtype = value.dtype
        shape = value.shape
        self.create(...

Reply:
<p>Updated to describe both explicit <code>create()</code>  method as well as creating from an existing array with simple assignment.  There are subtleties once we are using MPI, but I will address that in the parallel tutorial.</p>